### PR TITLE
additional model attributes

### LINKED ISSUE #5: [FEATURE] Additional attributes for vortex model
**Is your feature request related to a problem? Please describe.**
When using prediction pipeline, the behavior of vortex model is different compared to vortex runtime model. 

1. It doesnt have any 'input_specs' attribute
2. It required dataset to be downloaded to get the `class_names` attributes

Additionally, we also lack of information to resume interrupted training

So, we need this to be implemented

**Describe the solution you'd like**
Currently, .pth is directly stored the model's params. We can extend this to store a dictionary of model's attributes and training states

1. Adding input specs
2. Adding class_names
3. Adding current epoch and optimizer state
4. Update scheduler state with info from ckpt on resume

### PR DESCRIPTION
**Pull request by triwahyuu**
closes #5 

## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- close #5 

Related issues : 

## What is the new behavior?

- update checkpoint model format
- support for resume training in training pipeline
- reading `class_names` attribute from checkpoint in prediction and export pipeline
- change `class_names` in dataset to optional
- adding model checkpoint update script in `script/update_model.py`

### Checkpoint Model Format
default (required) member in checkpoint:
- `epoch` --> number of epochs the model have been trained
- `state_dict` -> trained model's state dict, this is the same as the entire data as the old format
- `optimizer_state` --> optimizer state in training
- `config` --> configuration used to train the model

additional member (if any):
- `class_names` --> model's output class names map
- `metrics` --> per epoch training metrics
- `scheduler_state` -> `state_dict` for trainer's lr scheduler

## Checklist

- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/vortex/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by triwahyuu** [comment]
ready to be merge cc @alphinside
